sdk/src/asset_handlers/tiff_io.rs

```diff
@@ -1610,15 +1617,22 @@ impl RemoteRefEmbed for TiffIO {
     ) -> Result<()> {
         match embed_ref {
             crate::asset_io::RemoteRefEmbedType::Xmp(manifest_uri) => {
-                #[cfg(feature = "xmp_write")]
-                {
-                    crate::embedded_xmp::add_manifest_uri_to_file(asset_path, &manifest_uri)
-                }
+                let output_buf = Vec::new();
+                let mut output_stream = Cursor::new(output_buf);
 
-                #[cfg(not(feature = "xmp_write"))]
+                // block so that source file is closed after embed
                 {
-                    Err(crate::error::Error::MissingFeature("xmp_write".to_string()))
+                    let mut source_stream = std::fs::File::open(asset_path)?;
+                    self.embed_reference_to_stream(
```

Comment:
I didn't update the embed_ref call to use the streaming version for JPEG, I should probably do that. It still sues xmp_toolkit if you have file_io enabled.